src/app/[locale]/components/FormProgress.tsx

```diff
@@ -0,0 +1,84 @@
+'use client'
+
+import React, { useEffect, useState } from 'react'
+import { Button, Heading } from '@utrecht/component-library-react'
+import { useTranslations } from 'next-intl'
+import { useStepperStore } from '@/store/stepper_store'
+import { useFormStore } from '@/store/form_store'
+import { steps, useRouter } from '@/routing/navigation'
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa'
+import { FormStep } from '@/types/form'
+
+const FormProgress = () => {
+  const t = useTranslations('stepper')
+  const {
+    step,
+    goToStep,
+    onNavToSummary,
+    visitedSteps,
+    resetVisitedSteps,
+    onGoBack,
+  } = useStepperStore()
+  const { resetForm } = useFormStore()
+  const router = useRouter()
+
+  const [percentage, setPercentage] = useState<number>(0)
+
+  useEffect(() => {
+    setPercentage((step / 4) * 100)
+  }, [step])
+
+  const resetState = () => {
+    resetVisitedSteps()
+    goToStep(FormStep.STEP_1_DESCRIPTION)
+    resetForm()
+    router.push(steps[FormStep.STEP_1_DESCRIPTION])
+  }
+
+  if (!visitedSteps.includes(FormStep.STEP_4_SUMMARY)) {
+    return (
+      <div className="relative flex flex-col-reverse sm:flex-col">
+        <div>
+          {step > FormStep.STEP_1_DESCRIPTION && (
+            <Button
+              appearance={'subtle-button'}
+              className="sm:absolute sm:left-0 sm:-top-2 stepper-button-hover pl-0-overwrite"
+              onClick={() => onGoBack(true)}
+            >
+              <FaChevronLeft />
+              {t('back')}
+            </Button>
+          )}
+
+          {step < FormStep.STEP_4_SUMMARY &&
+            visitedSteps.includes(FormStep.STEP_3_CONTACT) && (
```

Comment:
De visitedSteps logica zou denk ik wat gemakkelijker kunnen, in principe is het (als ik kijk naar waar visitedSteps state wordt gebruikt) niet noodzakelijk om te weten of je een bepaalde stap hebt bezocht, maar enkel welke stap je als laatst hebt bezocht. Dit kan dus dan iets worden als:

```
lastVisitedStep: FormStep
setLastVisitedStep: (step: FormStep) => void
```

Dan hoef je ook geen array lookups meer te doen met .includes en zou je dit kunnen vervangen met iets als:
```
step < FormStep.STEP_4_SUMMARY && lastVisitedStep === FormStep.STEP_3_CONTACT
```

Reply:
Dat werkt helaas niet. Als je vanaf de samenvattingspagina terug navigeert naar een van de eerder pagina's en vervolgens op volgende klikt wordt de 'lastVisitedStep' opnieuw bijgewerkt maar op een manier dat niet wenselijk is. Wat misschien nog zou kunnen is dat de waarde alleen opgehoogd wordt maar op zich werkt dit prima toch? en de array lookup is op een array van maximaal 4 waardes